project/frontend/src/components/LocationFinder.js

```diff
@@ -23,70 +49,78 @@ class LocationFinder extends React.Component {
    */
   getLocationFromGeolocate() {
     if (navigator.geolocation) {
-      const google = window.google;
-      const geocoder = new google.maps.Geocoder();
-      navigator.geolocation.getCurrentPosition((position) => {
-        const currLocation = {
-          lat: position.coords.latitude,
-          lng: position.coords.longitude,
-        };
-        geocoder.geocode({ location: currLocation }, (results, status) => {
-          if (status === 'OK') {
-            const locationName = results[0].formatted_address;
-            this.setState({ locationName });
-            this.setState({ submitted: true });
-            this.props.sendData({ currLocation, locationName });
-          } else {
-            alert(
-              'Geocode was not successful for the following reason: ' +
-                status +
-                '. Try entering a location in the text box below.'
-            );
-          }
-        });
-      });
+      navigator.geolocation.getCurrentPosition(
+        this.handleGeolocateSuccess,
+        this.handleGeolocateError
+      );
     } else {
       alert(
         'Cannot find your location. Try entering a location in the text box below.'
       );
     }
   }
 
-  /**
-   * Uses Geocoder to get location information based on user input.
-   */
-  getLocationFromText(event) {
-    event.preventDefault();
+  handleGeolocateSuccess(position) {
     const google = window.google;
     const geocoder = new google.maps.Geocoder();
-    geocoder.geocode({ address: this.state.userInput }, (results, status) => {
+    const currLocation = {
+      lat: position.coords.latitude,
+      lng: position.coords.longitude,
+    };
+    geocoder.geocode({ location: currLocation }, (results, status) => {
       if (status === 'OK') {
-        const currLocation = {
-          lat: results[0].geometry.location.lat(),
-          lng: results[0].geometry.location.lng(),
-        };
         const locationName = results[0].formatted_address;
         this.setState({ locationName });
         this.setState({ submitted: true });
+        this.setState({ error: false });
```

Comment:
same comment as Surbhi from above! I'm pretty sure you can do this in one setState call https://discuss.codecademy.com/t/can-this-setstate-change-multiple-properties-at-once/392713